rfc/rfc-2/2 - Deletion Info Conversion.md

```diff
@@ -0,0 +1,221 @@
+original data files. This deletion vector type is useful when identifying and storing the exact position of the row(s) 
+is expensive, but the content that identifies the row is clear. This feature is particularly useful in streaming use 
+cases where updates are frequent. Currently, Iceberg supports equality deletes, however, there is a proposal to replace 
+it in future versions.
+
+#### Position deletes
+Position deletes, on the other hand, specify the ordinals of records within a data file that should be considered 
+deleted. This method relies on identifying the exact position within the data file, thereby providing a more precise and
+direct approach to managing row-level deletions. Position deletes can be further categorized into two representations: 
+simple table representations and compressed bitmap representations.
+
+**_Simple table representations_** involve a straightforward structure typically stored in Parquet format, containing 
+columns like `file_path` and `pos`. The `file_path` indicates the path of the data file, while `pos` specifies the 
+ordinal of the record to be deleted. These files can optionally include the row data for equality deletes, aiding in 
+identifying the exact change which is particularly useful in CDC applications.
+
+A **_compressed bitmap representations_**, leverage a compressed bitmap to represent deletions. This method allows for a
+more compact and optimized storage of delete vectors, which in turn saves memory and network bandwidth. The 
+_RoaringBitmap_ is a popular choice for this representation. While the bitmaps are typically stored in separate files, 
+in cases where the bitmaps are small because the number of deleted records is low, they can be embedded within the 
+commit logs to optimize the number of network round trips required to load all the files.
+
+#### Row level deletes in Delta Lake
+Currently, Delta Lake supports RoaringBitmaps for deletion vectors, while Iceberg supports simple table representations.
+The java libraries of Delta Lake and Iceberg provide APIs to read and write between these representations. Delta Lake 
+provides support for deletion vectors for `DELETE` operations since versions 2.4, while `UPDATE` and `MERGE` commands 
+default to the copy-on-write mode.
+
+In Delta Lake, the implementation of deletion vectors involves `actions` in the commit logs, namely `AddFile` and 
+`RemoveFile`. When a DML command is executed, it identifies the rows to be modified and generates a deletion vector, 
+typically in a compressed bitmap format like RoaringBitmap. The generated deletion vector is stored directly 
+in the commit logs `json` files (when the number of deletions is minimal). When stored in separate files, the files 
+paths are encoded and referenced in the json files. With each DML operation, the commit log records the removal of the 
+original data file and its addition with updated details including the newly added deletion vector. 
+
+Example:
```

Comment:
https://github.com/delta-io/delta/blob/master/PROTOCOL.md#add-file-and-remove-file

Say some rows are updated in `file_a.parquet`. Wouldn't this result in one more row, that contains the new values for the deleted rows?
Delta spec doesn't describe this detail as well. Is the order of application determined by the order of actions in the log file?

Reply:
That's a great question! 
I haven't looked into the reference implementation that would resolve conflicting actions (add and remove in this case) for the same data file. Maybe the reference implementation requires the delete vectors descriptions to also match if two actions need to be equal. 
I agree that the delta spec should clarify this. I'm happy to take a look at it, but it would be best to confirm on the Delta Lake dev list as well.